mapie/tests/test_mondrian.py

```diff
@@ -0,0 +1,434 @@
+    model.fit(x, y)
+    mondrian = MondrianCP(
+        mapie_estimator=MapieClassifier(estimator=model, cv="prefit")
+    )
+    groups = np.random.choice(10, len(x) - 1)
+    with pytest.raises(ValueError, match=r".*he number of individuals in*"):
+        mondrian.fit(x, y, groups=groups)
+
+
+def test_all_groups_in_predict_are_in_fit():
+    """
+    Test that all groups in predict are in fit"""
+    x, y = TOY_DATASETS["classification"]
+    ml_model = ML_MODELS["classification"]
+    model = clone(ml_model)
+    model.fit(x, y)
+    mondrian = MondrianCP(
+        mapie_estimator=MapieClassifier(estimator=model, cv="prefit")
+    )
+    groups = np.random.choice(10, len(x))
+    mondrian.fit(x, y, groups=groups)
+    groups = np.array([99] * len(x))
+    with pytest.raises(ValueError, match=r".*There is at least one new*"):
+        mondrian.predict(x, groups=groups, alpha=.2)
+
+
+def test_groups_and_x_have_same_length_in_predict():
+    """
+    Test that groups and x have the same length in predict"""
+    x, y = TOY_DATASETS["classification"]
+    ml_model = ML_MODELS["classification"]
+    model = clone(ml_model)
+    model.fit(x, y)
+    mondrian = MondrianCP(
+        mapie_estimator=MapieClassifier(estimator=model, cv="prefit")
+    )
+    groups = np.random.choice(10, len(x))
+    mondrian.fit(x, y, groups=groups)
+    groups = np.random.choice(10, len(x) - 1)
+    with pytest.raises(ValueError, match=r".*The number of individuals in*"):
+        mondrian.predict(x, groups=groups, alpha=.2)
+
+
+def test_alpha_none_return_one_element():
+    """
+    Test that if alpha is None, the output is a single element"""
+    x, y = TOY_DATASETS["classification"]
+    ml_model = ML_MODELS["classification"]
+    model = clone(ml_model)
+    model.fit(x, y)
+    mondrian = MondrianCP(
+        mapie_estimator=MapieClassifier(estimator=model, cv="prefit")
+    )
+    groups = np.random.choice(10, len(x))
+    mondrian.fit(x, y, groups=groups)
+    preds = mondrian.predict(x, groups=groups)
+    assert len(preds) == len(x)
+
+
+def test_groups_is_list_ok():
+    """
+    Test that the groups can be a list"""
+    x, y = TOY_DATASETS["classification"]
+    ml_model = ML_MODELS["classification"]
+    model = clone(ml_model)
+    model.fit(x, y)
+    mondrian = MondrianCP(
+        mapie_estimator=MapieClassifier(estimator=model, cv="prefit")
+    )
+    groups = np.random.choice(10, len(x)).tolist()
+    mondrian.fit(x, y, groups=groups)
+    mondrian.predict(x, groups=groups, alpha=.2)
+
+
+@pytest.mark.parametrize("mapie_estimator_name", VALID_MAPIE_ESTIMATORS_NAMES)
+def test_same_results_if_only_one_group(mapie_estimator_name):
+    """
+    Test that the results are the same if there is only one group"""
+    task_dict = VALID_MAPIE_ESTIMATORS[mapie_estimator_name]
+    mapie_estimator = task_dict["estimator"]
+    mapie_kwargs = task_dict["kwargs"]
+    task = task_dict["task"]
+    x, y = TOY_DATASETS[task]
+    y = np.abs(y)
```

Comment:
Wherever we use ``np.abs(y)`` (in this file), we should check that we can remove it without any problem. If not, we should explain in the documentation why this is not possible.

Reply:
It's not possible because de Gamma Conformity Score requires to predict only positive values (which should be changed at some point I think btw)